drivers/modem/ublox-sara-r4.c

```diff
@@ -1258,6 +1349,41 @@ static int create_socket(struct modem_socket *sock, const struct sockaddr *addr)
 		return -1;
 	}
 
+	if (sock->ip_proto == IPPROTO_TLS_1_2) {
+		char buf[sizeof("AT+USECPRF=#,#,#######\r")];
+
+		snprintk(buf, sizeof(buf), "AT+USOSEC=%d,1,%d", sock->id, sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
```

Comment:
Would be good to verify the return value of `modem_cmd_send()`, I assume that failed transition of any of these commands results in an invalid socket to use.